---
layout: page
title: About Sean
permalink: /about/
---

Greetings! My name is Sean Maden, I am a Computational Biology Ph.D student at Oregion Health & Science University in Portland, OR. I work in Dr. Abhinav Nellore's Lab, in affiliation with PDXGx and Dr. Reid Thompson's Lab. I am a researcher and scholar of bioinformatics, epigenetics, and integrative computational oncology.

### Programming and Research Background

I specialize in compute systems for the study of epigenetics, oncolongy, and integration of various molecular data types. My work is highly collaborative, and I am constantly seeking new opportunities to expand digital workflows and systems to tackle reproducibility and cutting edge biomedical science. If you are interested in a collaboration, don't hesitate to reach out!

I have extensive programming background in R, especially applying Bioconductor resources. My work utilizes all manner of OS environments and systems, including remote server environments and batch management systems like slurm. My background includes specialization in study of molecular and epigenetic dynamics in gastroesophageal cancers using methylation arrays. I have since worked extensively with gene expression, variant sequencing, and various clinical data types. I excel at analysis workflows that include public data repositories, such as The Cancer Genome Atlas (TCGA) and Gene Expression Omnibus (GEO). I developed a few proto-R and shiny packages to aid in epigenetics analysis, including [methyPre](https://github.com/metamaden/methyPre) for methylation array preprocessing, [PrImiR](), a consensus miRNA database, [InfiniumPurifyR](https://github.com/metamaden/InfiniumPurifyR) an R port of a tissue bias correction method for methylation array data, and [cgmappeR](https://github.com/metamaden/cgmappeR), a GUI and Gviz wrapper for interactive sequence visualization. 

As a Ph.D candidate and researcher at the Knight Cancer Institute, I am designing a platform-specific auto-compilation technology to analyze all publically available samples from GEO, which I am coding with Python, Bash, and R. This technology utilizes extensive server functionality, including automated job queue management with streaming to a backend SQLite database, file versioning and management with a Mongo document database, and preprocessing functionality that uses MetaSRA-pipeline. 

Outside of my bioinformatics pursuits, I am passionate about data-driven lay press writing, hackathons, data activism, microbrews, board games, and bouldering.

### Curriculum Vitae in Science
Access my research CV from my dedicated [CV repo.](https://github.com/metamaden/CV_repo)

